app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+	has_many :todos
+
+	def authenticate(password)
+		hash = BCrypt::Engine.hash_secret(password, self.password_salt)
+		self.password_hash == hash	
+	end
+
+	def encrypt_password(password)
+		self.password_salt = BCrypt::Engine.generate_salt
+		self.password_hash = BCrypt::Engine.hash_secret(password, self.password_salt)
+	end
+
+end
```

Comment:
Hi Raymond, good stuff. We're using `Devise` gem for authentication. Do you know how to use it? This is good enough though.